nmtwizard/preprocess/consumer.py

```diff
@@ -90,7 +90,8 @@ def finalize(self):
 class SummaryLogger(Consumer):
     """A consumer that reduces operators filter information and logs the result."""
 
-    def __init__(self):
+    def __init__(self, sampler_summary):
+        self._sampler_summary = sampler_summary
         super().__init__()
```

Comment:
Prefer initializing the attribute after calling the parent constructor:

```suggestion
        super().__init__()
        self._sampler_summary = sampler_summary
```